drivers/gpio/gpio_keys_zephyr.c

```diff
@@ -192,8 +192,16 @@ static int gpio_keys_zephyr_get_pin(const struct device *dev, uint32_t idx)
 static int gpio_keys_init(const struct device *dev)
 {
 	struct gpio_keys_data *data = dev->data;
+	const struct gpio_keys_config *cfg = dev->config;
+	int ret;
 
 	for (int i = 0; i < data->num_keys; i++) {
+		ret = gpio_pin_configure_dt(&cfg->pin_cfg[i].spec, GPIO_INPUT);
```

Comment:
You should call `gpio_is_ready_dt` before this.

Reply:
Good point thanks, did not realize we had a specialized gpio+dt one for that, good stuff.